src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,30 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    this.hidden = false;
+
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(obj) {
+    if (obj instanceof Herbivore && obj.health > 0 && !obj.hidden) {
+      obj.health -= 50;
+    }
+    Animal.alive = Animal.alive.filter((animal) => animal.health !== 0);
```

Comment:
The filter condition should check for `animal.health > 0` instead of `animal.health !== 0` to ensure that only alive animals remain in the `Animal.alive` array.